opentelemetry-http/src/lib.rs

```diff
@@ -7,6 +7,10 @@ pub use bytes::Bytes;
 pub use http::{Request, Response};
 use opentelemetry::propagation::{Extractor, Injector};
 
+/// Helper for injecting headers into HTTP Requests. This is used for OpenTelemetry context
```

Comment:
From a quick glance, this can be moved out to its own crate, as this does not require any particular httpclient implementation, so can move quickly. 

Would it help anyone if we move this to own crate? I guess there are users who need this part (for context propagation over http), but does not need an exporting mechanism over http (as they can use alternatives like gRPC)?

Reply:
@TommyCpp suggested its better to keep crate count low, but we can achieve this via feature flags.. Will have to follow up separately for that.